### PR TITLE
Switch to yargs.

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

File: bigquery/system-test/queries.test.js [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 'use strict';
 
-var example = require('../query');
+var example = require('../queries');
```
Comment (ace-n) [inline bigquery/system-test/queries.test.js]: Not terribly worried about this change (because it needs to be done anyway), but this wasn't _technically_ a `yargs` change. :stuck_out_tongue:

File: logging/sinks.js [hunk 2]
```diff
@@ -181,24 +182,24 @@ cli
...
+    program.listSinks(utils.makeHandler(true, 'id'));
   })
   .command('update <name> <metadata>', 'Update the metadata for the specified sink.', {}, function (options) {
     try {
       options.metadata = JSON.parse(options.metadata);
     } catch (err) {
       return console.error('"metadata" must be a valid JSON string!');
     }
-    program.updateSink(options, console.log);
+    program.updateSink(utils.pick(options, ['name', 'metadata']), utils.makeHandler(false));
```
Comment (ace-n) [inline logging/sinks.js]: Another option (for later use): instead of _whitelisting_ the options we want, we _blacklist_ the ones we don't. (IIRC we briefly discussed this yesterday but didn't rule it out.)

File: pubsub/README.md [hunk 3]
```diff
@@ -61,22 +63,23 @@ View the [documentation][subscriptions_docs] or the [source code][subscriptions_
 __Usage:__ `node subscriptions --help`
 
 ```
-Usage: node subscriptions COMMAND [ARGS...]
-
 Commands:
+  create <topic> <name>  Create a new subscription.
+  list [topic]           List subscriptions.
+  pull <subscription>    Pull messages from the specified subscription.
+  delete <subscription>  Delete the specified dataset.
```
Comment (ace-n) [inline pubsub/README.md]: Is `dataset` a typo?

File: pubsub/subscriptions.js [hunk 4]
```diff
@@ -157,50 +157,44 @@ function pullMessages (subscriptionName, callback) {
...
+  .command('create <topic> <name>', 'Create a new subscription.', {}, function (options) {
+    program.createSubscription(options.topic, options.name, makeHandler(true, 'id'));
+  })
+  .command('list [topic]', 'List subscriptions.', {}, function (options) {
+    program.listSubscriptions(options.topic, makeHandler(true, 'id'));
+  })
+  .command('pull <subscription>', 'Pull messages from the specified subscription.', {}, function (options) {
+    program.pullMessages(options.subscription, makeHandler(false));
+  })
+  .command('delete <subscription>', 'Delete the specified dataset.', {}, function (options) {
```
Comment (ace-n) [inline pubsub/subscriptions.js]: Typo -`s/dataset/subscription/`

File: pubsub/test/subscriptions.test.js [hunk 5]
```diff
@@ -188,57 +188,57 @@ describe('pubsub:subscriptions', function () {
...
+      assert.equal(program.deleteSubscription.calledOnce, true);
+      assert.deepEqual(program.deleteSubscription.firstCall.args.slice(0, -1), [subscriptionName]);
+    });
 
-      sinon.stub(program, 'delete');
-      program.main(['delete']);
-      assert(program.delete.calledOnce);
+    it('should call listSubscriptions', function () {
+      var program = getSample().program;
+      sinon.stub(program, 'listSubscriptions');
```
Comment (ace-n) [inline pubsub/test/subscriptions.test.js]: Is there any difference/preference between
```javascript
sinon.stub(program, 'listSubscriptions');
```
and
```javascript
program.listSubscriptions = sinon.stub();
```

If not, should we stick to one of these idioms?

File: pubsub/test/topics.test.js [hunk 6]
```diff
@@ -96,7 +96,7 @@ describe('pubsub:topics', function () {
       var sample = getSample();
       var error = 'error';
       sample.mocks.topic.delete.callsArgWith(0, new Error(error));
-      sample.program.delete(topicName, function (err) {
+      sample.program.deleteTopic(topicName, function (err) {
```
Comment (ace-n) [inline pubsub/test/topics.test.js]: These renaming changes are necessary - but they don't seem to be related to `yargs`. :stuck_out_tongue:

File: storage/test/encryption.test.js [hunk 7]
```diff
@@ -177,31 +177,45 @@ describe('storage:encryption', function () {
...
 
     it('should call uploadEncryptedFile', function () {
       var program = getSample().program;
 
       sinon.stub(program, 'uploadEncryptedFile');
       program.main(['upload', bucketName, fileName, fileName, key]);
-      assert(program.uploadEncryptedFile.calledOnce);
+      assert.equal(program.uploadEncryptedFile.calledOnce, true);
+      assert.deepEqual(program.uploadEncryptedFile.firstCall.args.slice(0, -1), [{
+        bucket: bucketName,
```
Comment (ace-n) [inline storage/test/encryption.test.js]: Is it worth putting this into a `var`? (Maybe something inserted at line 175?)

Comment (ace-n) [inline storage/test/encryption.test.js]: (On second thought, never mind.)

File: storage/test/files.test.js [hunk 8]
```diff
@@ -449,71 +449,109 @@ describe('storage:files', function () {
...
       var program = getSample().program;
 
       sinon.stub(program, 'moveFile');
       program.main(['move', bucketName, fileName, fileName]);
-      assert(program.moveFile.calledOnce);
+      assert.equal(program.moveFile.calledOnce, true);
+      assert.deepEqual(program.moveFile.firstCall.args.slice(0, -1), [{
+        bucket: bucketName,
+        srcFile: fileName,
+        destFile: fileName
```
Comment (ace-n) [inline storage/test/files.test.js]: Should these `fileName`s be different? (To make sure we don't mess up the order of `srcFile` and `destFile`.)

File: storage/test/files.test.js [hunk 9]
```diff
@@ -449,71 +449,109 @@ describe('storage:files', function () {
...
 
     it('should call copyFile', function () {
       var program = getSample().program;
 
       sinon.stub(program, 'copyFile');
       program.main(['copy', bucketName, fileName, bucketName, fileName]);
-      assert(program.copyFile.calledOnce);
+      assert.equal(program.copyFile.calledOnce, true);
+      assert.deepEqual(program.copyFile.firstCall.args.slice(0, -1), [{
+        srcBucket: bucketName,
```
Comment (ace-n) [inline storage/test/files.test.js]: Bucket/file names - see comment above.

File: storage/test/transfer.test.js [hunk 10]
```diff
@@ -485,63 +485,90 @@ describe('storage:transfer', function () {
 
       sinon.stub(program, 'createTransferJob');
       program.main(['jobs', 'create', bucketName, bucketName, 'time', 'date']);
-      assert(program.createTransferJob.calledOnce);
+      assert.equal(program.createTransferJob.calledOnce, true);
+      assert.deepEqual(program.createTransferJob.firstCall.args.slice(0, -1), [{
```
Comment (ace-n) [inline storage/test/transfer.test.js]: Bucket names - should they be different?

File: pubsub/test/subscriptions.test.js (hunk 5 above)
Comment (jmdobry) [inline pubsub/test/subscriptions.test.js]: The former allows you to restore the original function if desired. With the second the original is gone.

File: pubsub/README.md (hunk 3 above)
Comment (jmdobry) [inline pubsub/README.md]: Fixed.

File: pubsub/subscriptions.js (hunk 4 above)
Comment (jmdobry) [inline pubsub/subscriptions.js]: Fixed.

File: translate/system-test/translate.test.js [hunk 11]
```diff
@@ -58,7 +58,7 @@ describe('translate:translate', function () {
       program.translateText(options, function (err, translation) {
         assert.ifError(err);
         assert.equal(translation, expected);
-        assert(console.log.calledWith('Translated text to %s', 'Russian'));
+        assert(console.log.calledWith('Translated text to %s:', 'Russian'));
```
Comment (ace-n) [inline translate/system-test/translate.test.js]: Technically unrelated - but these are fine.

**Comment by ace-n** [comment]
LGTM, other than the nits that I mentioned.

File: storage/test/transfer.test.js (hunk 10 above)
Comment (jmdobry) [inline storage/test/transfer.test.js]: Fixed.

File: storage/test/files.test.js (hunk 9 above)
Comment (jmdobry) [inline storage/test/files.test.js]: Fixed.

File: storage/test/files.test.js (hunk 8 above)
Comment (jmdobry) [inline storage/test/files.test.js]: Fixed.

File: logging/sinks.js (hunk 2 above)
Comment (jmdobry) [inline logging/sinks.js]: Will address in another PR.

File: storage/test/transfer.test.js [hunk 12]
```diff
@@ -15,6 +15,7 @@
 
 var proxyquire = require('proxyquire').noCallThru();
 var bucketName = 'foo';
```
Comment (ace-n) [inline storage/test/transfer.test.js]: Nit: should this be `srcBucketName`?

File: storage/test/files.test.js [hunk 13]
```diff
@@ -15,7 +15,9 @@
 
 var proxyquire = require('proxyquire').noCallThru();
 var bucketName = 'foo';
-var fileName = 'test.txt';
+var fileName = 'test1.txt';
```
Comment (ace-n) [inline storage/test/files.test.js]: Nit: should this be `srcFileName`? (To clearly differentiate it vs. the other `*FileName` variables.)

File: storage/test/transfer.test.js (hunk 12 above)
Comment (jmdobry) [inline storage/test/transfer.test.js]: Fixed.

File: storage/test/files.test.js (hunk 13 above)
Comment (jmdobry) [inline storage/test/files.test.js]: Fixed.

**Comment by jmdobry** [comment]
@dpebot Merge when green.
